File: ios/valens/ScreenSecureModule.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ScreenSecure, RCTEventEmitter)

RCT_EXTERN_METHOD(setSecure:(BOOL)enabled
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isSecure:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/valens/ScreenSecureModule.swift
import UIKit
import React

@objc(ScreenSecure)
class ScreenSecureModule: RCTEventEmitter {
  private var screenshotObserver: NSObjectProtocol?

  @objc override static func requiresMainQueueSetup() -> Bool {
    true
  }

  override func supportedEvents() -> [String]! {
    ["UserDidTakeScreenshot"]
  }

  override func startObserving() {
    guard screenshotObserver == nil else { return }

    screenshotObserver = NotificationCenter.default.addObserver(
      forName: UIApplication.userDidTakeScreenshotNotification,
      object: nil,
      queue: .main,
    ) { [weak self] _ in
      self?.sendEvent(withName: "UserDidTakeScreenshot", body: nil)
    }
  }

  override func stopObserving() {
    if let screenshotObserver {
      NotificationCenter.default.removeObserver(screenshotObserver)
      self.screenshotObserver = nil
    }
  }

  @objc func setSecure(
    _ enabled: Bool,
    resolve: @escaping RCTPromiseResolveBlock,
    reject: @escaping RCTPromiseRejectBlock,
  ) {
    // iOS uses JS blur + screenshot listener; native secure layer stays off.
    resolve(true)
  }

  @objc func isSecure(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject: @escaping RCTPromiseRejectBlock,
  ) {
    resolve(false)
  }
}
